cluster-autoscaler/core/placeholderpod/create.go

```diff
@@ -0,0 +1,68 @@
+package placeholderpod
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// CreateReplicaSets create one or more replica set(s) of placeholder pods
+func CreateReplicaSets(sets ...ReplicaSet) []*apiv1.Pod {
+	pods := []*apiv1.Pod{}
+	for _, set := range sets {
+		for i := int64(0); i < set.Count; i++ {
+			podSpec := set.PodSpec
+			nodeStickiness := podSpec.NodeStickiness
+			milliCPU := podSpec.MilliCPU
+			memory := podSpec.Memory
+			req := apiv1.ResourceList{}
+			req[apiv1.ResourceCPU] = *resource.NewMilliQuantity(milliCPU, resource.DecimalSI)
+			req[apiv1.ResourceMemory] = *resource.NewQuantity(memory, resource.DecimalSI)
+			var name string
+			if set.Count > 1 {
+				name = fmt.Sprintf("%s-%d", set.Name, i)
+			} else {
+				name = set.Name
+			}
+			pod := &apiv1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              fmt.Sprintf("placeholder-%s-%dm-%dMB", name, milliCPU, memory/1024/1024),
+					CreationTimestamp: metav1.NewTime(time.Now()),
+					// Without a namespace, this causes an error like:
+					// > POST https://10.3.0.1:443/api/v1/namespaces/default/events 422 Unprocessable Entity in 1 milliseconds
+					// due to an invalid pod:
+					// > 'Event "placeholder-12-2800m-5626449100.14bd31ca98a9fa02" is invalid: involvedObject.namespace: Required value: required for kind Pod' (will not retry!)
+					Namespace: "default",
+				},
+				Spec: apiv1.PodSpec{
```

Comment:
The main problem with antiaffinity is that it works only with caching. Otherwise it is a performance disaster. And we don't use affinity caching in CA as we run various hypothetical simulations. 

Please create a document as requested in my last comment and we will proceed the discussion there.